app/components/EmojiPicker.tsx

```diff
@@ -0,0 +1,95 @@
+import data from "@emoji-mart/data";
+import Picker from "@emoji-mart/react";
+import { CloseIcon } from "outline-icons";
+import * as React from "react";
+import {
+  usePopoverState,
+  Popover as ReakitPopover,
+  PopoverDisclosure,
+  PopoverArrow,
+  PopoverProps,
+} from "reakit/Popover";
+import styled from "styled-components";
+import { depths } from "@shared/styles";
+import NudeButton from "~/components/NudeButton";
+import Tooltip from "~/components/Tooltip";
+
+type Props = PopoverProps & {
+  disclosure: any;
+  children: React.ReactNode;
+  popoverState?: any;
+  tabIndex?: number;
+  showPopoverArray?: boolean;
+};
+
+const Popover: React.FC<Props> = ({
+  disclosure,
+  children,
+  popoverState,
+  showPopoverArray,
+  ...rest
+}) => {
+  const popover = usePopoverState(popoverState);
+
+  return (
+    <>
+      <PopoverDisclosure
+        {...popover}
+        ref={disclosure.ref}
+        {...disclosure.props}
+      >
+        {(disclosureProps) => React.cloneElement(disclosure, disclosureProps)}
+      </PopoverDisclosure>
+      <BasePopover {...popover} {...rest}>
+        {showPopoverArray && <PopoverArrow {...popover} />}
+        {children}
+      </BasePopover>
+    </>
+  );
+};
```

Comment:
IMO the existing [`Popover`](https://github.com/outline/outline/blob/main/app/components/Popover.tsx) isn't sufficiently abstracted, revealing certain limitations. For example, it doesn't allow _injecting_ a custom `PopoverDisclosure` as opposed to this one. It'd be better if we can replace the existing one with this new abstraction. Leaving it here for now to contain the scope of current PR. It could be taken up separately. Thoughts on this @tommoor ?

Reply:
Yea, I can be into this – allows keeping the popover state out of the parent component